docs/pages/setup/guides/joining-nodes-aws-iam.mdx

```diff
@@ -0,0 +1,212 @@
+---
+title: Joining Nodes via AWS IAM Role
+description: Use the IAM join method to add nodes to your Teleport cluster on AWS
+---
+
+This guide will explain how to use the **IAM join method** to configure Teleport
+nodes and Proxies to join your Teleport cluster without sharing any secrets when
+they are running in AWS.
+
+The IAM join method is available in Teleport 8.3+. It is available to any
+Teleport node or Proxy running anywhere with access to IAM credentials, such as
+an EC2 instance with an attached IAM role. No specific permissions or IAM policy
+is required: an IAM role with no attached policies is sufficient. No IAM
+credentials are required on the Teleport Auth Service.
+
+<Details 
+scope={["oss", "enterprise"]} 
+scopeOnly
+opened
+title="Other AWS node joining methods"
+>
+
+There are two additional methods you can use to join your nodes to a Teleport
+cluster.
+
+The **EC2 join method** is available in self-hosted verisons of Teleport 7.3+.
+It is available to any Teleport node or Proxy running on an EC2 instance. Only
+one Teleport node or Proxy per EC2 instance may use the EC2 join method.
+
+IAM credentials with `ec2:DescribeInstances` permissions are required on
+your Teleport Auth Service. No IAM credentials are required on the nodes or
+Proxies.
+
+You can also configure nodes running in AWS to join a cluster via **secret tokens**,
+which is useful when you don't want to rely on AWS-specific APIs.
+Read more in the following guide:
+[Adding nodes to the cluster](../admin/adding-nodes.mdx)
+
+</Details>
+<Details 
+scope={["cloud"]} 
+scopeOnly
+opened
+title="Another AWS node joining method"
+>
+
+You can also configure nodes running in AWS to join a cluster via **secret tokens**,
+which is useful when you don't want to rely on AWS-specific APIs.
+Read more in the following guide:
+[Adding nodes to the cluster](../admin/adding-nodes.mdx)
+
+</Details>
+
+<Notice type="warning">
+The IAM join method is not compatible with the `--insecure-no-tls` flag, and is
+currently not supported in the AWS China or GovCloud partitions.
```

Comment:
I was going to make an update here but it might be best to roll it into this PR to avoid conflicts?

The mention of the `--insecure-no-tls` flag is not quite accurate, the issue is that when nodes are connecting in IoT mode TLS must be terminated at the proxy, not at an ALB, and `--insecure-no-tls` may not necessarily be set in this case. This is only relevant to self-hosted clusters when the node joins in IoT mode.

So I would probably word it:
```suggestion
The IAM join method will not work if TLS is terminated at a load balancer in front of your Teleport Proxy, unless the node is connecting directly to the Auth server.

The IAM join method is currently not support in the AWS China or GovCloud partitions.
```